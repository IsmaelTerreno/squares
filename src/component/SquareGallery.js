import React, { Component } from 'react';
import PropTypes from 'prop-types';
import PreviewSquare from './PreviewSquare';
import Loading from '../component/Loading';
import GalleryActions from '../container/GalleryActions.container';

class SquareGallery extends Component {

  static propTypes = {
    squares: PropTypes.arrayOf(PropTypes.shape({
      color: PropTypes.string,
      border: PropTypes.number,
      size: PropTypes.number,
      id: PropTypes.string
    })).isRequired,
    onClickSquare: PropTypes.func.isRequired,
    onLoad: PropTypes.func,
    isFetching: PropTypes.bool
  };

  componentDidMount(){
    this.props.onLoad();
  }
  render() {
    const { squares, onClickSquare, history, isFetching } = this.props;
    return (
      <div className="">
        <h4>Square Gallery</h4>
        <GalleryActions/>
        <div key={'button-bar-top'} className="dropdown-divider space" />
        { isFetching && <Loading /> }
        {squares.map((item, idx) => {
          return (
            <div className="square-preview-item" key={'square-preview-' + idx}
            onClick={() => {
                onClickSquare(item);
                history.push('/square');
              }
            } >
              <PreviewSquare
                color={item.color}
                border={item.border}
                size={item.size}
              />
              <div key={'square-preview-space' + idx} className="dropdown-divider space" />
            </div>
          );
        })}
      </div>
    );
  };

}

export default SquareGallery;
