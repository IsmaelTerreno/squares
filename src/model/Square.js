import uuid from 'uuid/v1';
/**
 * Square entity.
 */
class Square {
  constructor(idX, colorX, borderX, sizeX) {
    this.color = colorX;
    this.border = borderX;
    this.size = sizeX;
    this.id = idX || uuid();
  }
}

export default Square;
