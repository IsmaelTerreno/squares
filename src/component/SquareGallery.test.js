import React from 'react';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from "enzyme-adapter-react-16";
import SquareGallery from './SquareGallery';
import Square from '../model/Square';

configure({ adapter: new Adapter() });

const squareListMock = [
  new Square("1595a74a-c1da-11e8-a355-529269fb1459","#9ef094", 40, 15),
  new Square("2abf5ae4-c1da-11e8-a355-529269fb1459","#cd8af0", 40, 10),
  new Square("3afc81f2-c1da-11e8-a355-529269fb1459","#f09849", 40, 5)
];

describe('SquareGallery', () => {

  test('Should renders appropriately', () => {
    const mockOnClickSquare = jest.fn();
    const mockOnLoad = jest.fn();
    const wrapper = shallow(
      <SquareGallery
        squares={squareListMock}
        onClickSquare={mockOnClickSquare}
        onLoad={mockOnLoad}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('Should call the function passed via onClickSquare and onLoad props.', () => {
    const onClickSquareMock = jest.fn();
    const onLoadMock = jest.fn();
    const pushHistoryMock = jest.fn();
    const historyMock = { push: pushHistoryMock };
    const wrapper = shallow(
      <SquareGallery
        squares={squareListMock}
        onClickSquare={onClickSquareMock}
        onLoad={onLoadMock}
        history={historyMock}
      />
    );
    const squareItem = wrapper.find('.square-preview-item').first();
    squareItem.simulate('click');
    expect(onClickSquareMock).toHaveBeenCalled();
    expect(onLoadMock).toHaveBeenCalled();
    expect(pushHistoryMock).toHaveBeenCalled();
  });

});
