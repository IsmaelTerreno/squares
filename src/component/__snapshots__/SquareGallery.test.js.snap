// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SquareGallery Should renders appropriately 1`] = `
<div
  className=""
>
  <h4>
    Square Gallery
  </h4>
  <Connect(withRouter(GalleryActions)) />
  <div
    className="dropdown-divider space"
    key="button-bar-top"
  />
  <div
    className="square-preview-item"
    key="square-preview-0"
    onClick={[Function]}
  >
    <PreviewSquare
      border={40}
      color="#9ef094"
      size={15}
    />
    <div
      className="dropdown-divider space"
      key="square-preview-space0"
    />
  </div>
  <div
    className="square-preview-item"
    key="square-preview-1"
    onClick={[Function]}
  >
    <PreviewSquare
      border={40}
      color="#cd8af0"
      size={10}
    />
    <div
      className="dropdown-divider space"
      key="square-preview-space1"
    />
  </div>
  <div
    className="square-preview-item"
    key="square-preview-2"
    onClick={[Function]}
  >
    <PreviewSquare
      border={40}
      color="#f09849"
      size={5}
    />
    <div
      className="dropdown-divider space"
      key="square-preview-space2"
    />
  </div>
</div>
`;
